### PR TITLE
feat: Puzzle notification bar

### PR DESCRIPTION
**Pull request by aronshamash**
- as per https://nidigitalsolutions.jira.com/browse/REPLAT-5406

### Mobile:
Screenshot 2019-03-21 at 12 18 16

### Tablet
Screenshot 2019-03-21 at 12 17 59

### CODE REVIEW

**Comment by tuncaulubilge** [comment]
No need for a special logic to hide in Android @aronshamash . We'll just hide it if recentlyOpenedPuzzleCount is 0, which will be the case in android

File: packages/section/src/section.js [hunk 1]
```diff
@@ -46,6 +49,12 @@ class Section extends Component {
 
     return (
       <Responsive>
+        {isPuzzle && (
```
Comment (tuncaulubilge) [inline packages/section/src/section.js]: this should be a ListHeaderComponent in the flatlist:
```
          ListHeaderComponent={
            isPuzzle ? <PuzzleBar onPress={onPuzzleBarPress} /> : null
          }
```

File: packages/section/src/styles/index.js [hunk 2]
```diff
@@ -9,6 +9,22 @@ const styles = StyleSheet.create({
   listItemSeparator: {
     backgroundColor: colours.functional.keyline,
     height: 1
+  },
+  puzzleNotificationBarArrow: {
```
Comment (tuncaulubilge) [inline packages/section/src/styles/index.js]: Can we just call this a `puzzleBar`? Not a big deal but it's quite a long name atm :)

Comment (aronshamash) [reply packages/section/src/styles/index.js]: it is long

File: packages/section/src/puzzle-bar.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Text, View } from "react-native";
+import PropTypes from "prop-types";
+import { IconForwardArrow } from "@times-components/icons";
+import { colours } from "@times-components/styleguide";
+import styles from "./styles";
+
+const PuzzleBar = ({ count, onPress }) =>
```
Comment (tuncaulubilge) [inline packages/section/src/puzzle-bar.js]: We need some snapshots on this, then it's good to ship

File: packages/section/src/puzzle-bar.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  count ? (
+    <View style={styles.puzzleBarContainer}>
+      <Text onPress={onPress} style={styles.puzzleBarText}>
+        {count} recently opened puzzle
+        {count > 1 && "s"}
```
Comment (Ivaylo1987) [inline packages/section/src/puzzle-bar.js]: I like that 👍

File: packages/section/src/puzzle-bar.js (hunk 3 above)
Comment (aronshamash) [reply packages/section/src/puzzle-bar.js]: sorted

File: packages/section/__tests__/ios/__snapshots__/section.test.js.snap [hunk 5]
```diff
@@ -1,5 +1,80 @@
...
+      Object {
+        "color": "#1D1D1B",
+        "fontFamily": "TimesModern-Regular",
+        "fontSize": 20,
+      }
+    }
+  >
+    3
+     recently opened puzzle
+    s
```
Comment (tuncaulubilge) [inline packages/section/__tests__/ios/__snapshots__/section.test.js.snap]: Should this be in three lines?

Comment (aronshamash) [reply packages/section/__tests__/ios/__snapshots__/section.test.js.snap]: It doesn't do any harm. Think it is due to the formatting of the JSX.

File: packages/section/__tests__/shared.js [hunk 6]
```diff
@@ -45,11 +50,25 @@ export default () => {
...
           onPuzzlePress={() => {}}
           publicationName="TIMES"
+          recentlyOpenedPuzzleCount={1}
           section={edition.sections[0]}
         />
       ).toJSON()
     ).toMatchSnapshot();
   });
+
+  it("puzzle bar with one puzzle", () => {
```
Comment (nehasri89) [inline packages/section/__tests__/shared.js]: is having 1 or more than 1 count account for different states?
should there be a test with zero puzzle count?

File: packages/section/src/section.js [hunk 7]
```diff
@@ -57,6 +60,14 @@ class Section extends Component {
             ))
           }
           keyExtractor={item => item.elementId}
+          ListHeaderComponent={
+            isPuzzle ? (
```
Comment (nehasri89) [inline packages/section/src/section.js]: do we display the bar even with zero count?

File: packages/section/__tests__/shared.js (hunk 6 above)
Comment (aronshamash) [reply packages/section/__tests__/shared.js]: it is needed to test for the logic in the `PuzzleBar` : `{count > 1 && "s"}`

File: packages/section/src/section.js (hunk 7 above)
Comment (aronshamash) [reply packages/section/src/section.js]: no but i was missing a test for that. I have added now.

**Pull request merged**